oak_functions/client/rust/src/rekor.rs

```diff
@@ -159,10 +206,85 @@ pub fn verify_rekor_signature(
     pem_encoded_public_key_bytes: &[u8],
 ) -> anyhow::Result<()> {
     let signature_bundle = rekor_signature_bundle(log_entry_bytes)?;
+
+    verify_signature(
+        &signature_bundle.base64_signature,
+        &signature_bundle.canonicalized,
+        pem_encoded_public_key_bytes,
+    )
+    .context("failed to verify signedEntryTimestamp of the Rekor LogEntry")
+}
+
+/// Verifies the signature in the `body` over the `endorsement_bytes`, using the public key in
+/// `pem_encoded_public_key_bytes`.
+///
+/// Returns `Ok(())` if the verification succeeds, otherwise returns `Err()`.
+pub fn verify_rekor_body(
+    body: &Body,
+    endorsement_bytes: &[u8],
+    pem_encoded_public_key_bytes: &[u8],
+) -> anyhow::Result<()> {
+    if body.spec.signature.format != "x509" {
+        anyhow::bail!(
+            "unsupported signature format: {}; only x509 is supported",
+            body.spec.signature.format
+        )
+    }
+
+    // For now, we only support `sha256` as the hashing algorithm.
+    if body.spec.data.hash.algorithm != "sha256" {
+        anyhow::bail!(
+            "unsupported hash algorithm: {}; only sha256 is supported",
+            body.spec.data.hash.algorithm
+        )
+    }
+
+    // Check that hash of the endorsement statement matches the hash of the data in the Body.
+    let endorsement_hash = get_sha256(endorsement_bytes);
+    let endorsement_hash_hex = hex::encode(endorsement_hash);
+    if endorsement_hash_hex != body.spec.data.hash.value {
+        anyhow::bail!(
+            "the hash of the endorsement file ({:?}) does not match the hash of the data in the body of the rekor entry ({:?})",
+            endorsement_hash_hex,
+            body.spec.data.hash.value
+        )
+    }
+
+    // Check that the public key in the body matches the given public key. This in fact checks the
+    // consistency of the Rekor LogEntry, and we expect these public keys to always be the same.
+    let public_key_bytes = base64::decode(body.spec.signature.public_key.content.as_bytes())
+        .expect("failed to base64-decode the public key bytes in the Rekor LogEntry body");
+    if compare_keys(&public_key_bytes, pem_encoded_public_key_bytes)? != Ordering::Equal {
+        anyhow::bail!(
+            "the given public key is different from the public key in the rekor entry: {:?} vs. {:?}",
+            public_key_bytes,
+            pem_encoded_public_key_bytes,
+        )
+    }
+
+    verify_signature(
+        body.spec.signature.content.as_str().as_bytes(),
+        endorsement_bytes,
+        pem_encoded_public_key_bytes,
+    )
+    .context("failed to verify signature over the endorsement file")
+}
+
+/// Verifies the given base64-encoded signature over the given data bytes, using the given
+/// PEM-encoded public key.
+///
+/// Returns `Ok(())` if the verification succeeds, otherwise returns `Err()`.
```

Comment:
Just to throw it out there: you could change the return to Result<bool> and return false if the last step (`verify`) did not succeed. (because I kind of expected verify to return a bool, but I understand your choice, too)